snuba/datasets/discover.py

```diff
@@ -47,7 +47,7 @@ def detect_table(
     if conditions:
         for idx, condition in enumerate(conditions):
             if is_condition(condition):
-                if tuple(condition) == ("type", "!=", "transaction"):
+                if tuple(condition) == ("type", "=", "error"):
```

Comment:
I think there are other event types that need to go to the events table (csp, default, etc.) What's the reason for changing this condition ?

Reply:
We would either need cases for csp, security and default types or revert this change.

Reply:
I changed this for two reasons:

1) The check is broken. We will never match `("type", "!=", "transaction")` because the condition sent from sentry looks like `[[["isNull",["type"]],"=",1],["type","!=","transaction"]]`. 
2) It is useful to have some guarantee which dataset discover queries ultimately run against. Right now this works with transactions but not errors, so having this here actually fixes a number of queries that have `event.type:error` but are not running against the events dataset.

Also, this is a short circuit check for when people specify a specific event. If they specify an event type not in this condition, it will just continue on using the checks below. We don't need new cases here.

Reply:
" this is a short circuit check for when people specify a specific event. If they specify an event type not in this condition, it will just continue on using the checks below"

I don't know whether the intention of this condition is such. There are a few other conditions between this one and the final return EVENTS. That seems fine to me but I'll let @markstory  confirm. Anyway, since the condition you mention is actually broken today, I guess this does not matter.

Reply:
If this condition is not working, how does the the condition on line 52 work?

Reply:
The `!=` condition works differently from the `=` condition. We don't add the isNull check to plain `=` conditions.

Reply:
ok, this is not going to make things worse or more broken, though I would strongly advise for this logic to be revisited be more robust, either now, based on the old query structure, or after the AST is rolled out.